carcara/src/checker/rules/strings.rs

```diff
@@ -161,6 +188,50 @@ pub fn concat_unify(
     assert_eq(&expected, &expanded)
 }
 
+pub fn concat_conflict(
+    RuleArgs {
+        premises,
+        args,
+        conclusion,
+        pool,
+        polyeq_time,
+        ..
+    }: RuleArgs,
+) -> RuleResult {
+    assert_num_premises(premises, 1)?;
+    assert_num_args(args, 1)?;
+    assert_clause_len(conclusion, 1)?;
+
+    let term = get_premise_term(&premises[0])?;
+    let rev = args[0].as_term()?.as_bool_err()?;
+
+    if conclusion[0].as_bool_err()? {
+        return Err(CheckerError::ExpectedFalseBoolConstant(
+            conclusion[0].clone(),
+        ));
```

Comment:
There already is an error variant that covers this case, `ExpectedBoolConstant`:
```suggestion
        return Err(CheckerError::ExpectedBoolConstant(
            false,
            conclusion[0].clone(),
        ));
```